### ISSUE TITLE
Fastlane still searching for the Xcode beta path despite changing it.

### ISSUE THREAD

**Issue by AndyIbanez** [fastlane/fastlane#10402]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I have the latest, App Store release of Xcode and I'm trying to upload an app to TestFlight. However TestFlight still checks for the Beta version of Xcode for some reason.

I have done the following to try to fix this:

1. Ran `xcode-select -s /Applications/Xcode` and various equivalents.
2. Opened Xcode > Preferences > Locations and checked that "Command Line Tools" does not point to the old beta installation. It's currently pointing to the last official release.

Where else should I check for this? My `.bash_profile` doesn't have this path either.

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

Andys-MacBook-Pro:Mignori andyibanez$ fastlane beta
[20:57:52]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[20:57:55]: -------------------------------------------------
[20:57:55]: --- Step: Verifying required fastlane version ---
[20:57:55]: -------------------------------------------------
[20:57:55]: Your fastlane version 2.58.0 matches the minimum requirement of 2.28.7  ✅
[20:57:55]: --------------------------
[20:57:55]: --- Step: xcode_select ---
[20:57:55]: --------------------------

[!] Path '/Applications/Xcode-beta.app' doesn't exist

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.13.5 (Apple Git-94)           |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.12.6 (16G29)                    |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.0                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.28.7"
xcode_select "/Applications/Xcode-beta.app"
# git_url "git@bitbucket.org:fairese/ios-certificates.git"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    cocoapods

  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    # itc_team_id "613506"
    # match(type: "appstore") # more information: https://codesigning.guide
    match(force: true, type: "appstore", git_branch: "sergio-andres-ibanez-kautsch", username: "atanacross@gmail.com", app_identifier: ["com.fairese.Mignori"], git_url: "git@bitbucket.org:fairese/ios-certificates.git")
    scan(scheme: "Mignori")
    gym(export_method: "app-store", scheme: "Mignori") # Build your app - more options available
    pilot(skip_submission: true)

    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end

  desc "Deploy a new version to the App Store"
  lane :release do
    # match(type: "appstore")
    # snapshot
    match(type: "appstore", git_branch: "sergio-andres-ibanez-kautsch", username: "atanacross@gmail.com", app_identifier: ["com.fairese.Mignori"])
    scan(scheme: "Mignori")
    gym(scheme: "Mignori") # Build your app - more options available
    deliver(force: true, submit_for_review: true)
    # frameit
  end

  desc "Development environment"
  lane :dev_env do
    match(force: true, force_for_new_devices: true, type: "development", git_branch: "sergio-andres-ibanez-kautsch", username: "atanacross@gmail.com", app_identifier: ["com.fairese.Mignori"])
  end

  desc "Distribution environment"
  lane :dis_env do
    match(type: "appstore", git_branch: "sergio-andres-ibanez-kautsch", username: "atanacross@gmail.com", app_identifier: ["com.fairese.Mignori"])
  end

  desc "Frame Screenshots"
  lane :frame do
    frameit
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "com.fairese.Mignori" # The bundle identifier of your app
apple_id "atanacross@gmail.com" # Your Apple email address

team_id "7X7FABXK4C" # Developer Portal Team ID" # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.58.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.6        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| uber                      | 0.0.15       |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.4       |
</details>

*generated on:* **2017-09-22**
</details>

**Comment by AndyIbanez** [comment]
Fixed, Mr. Bot.

**Comment by mgrebenets** [comment]
@AndyIbanez 

In your description you said you tried `xcode-select -s /Applications/Xcode` and _various equivalents_.
First thing that I noticed that you are missing `.app` part in `Xcode.app`, also, I believe you are missing the `/Contents/Developer` part too. Or have you tried it as one of the variations you mentioned.

The correct command would be `xcode-select -s /Applications/Xcode.app/Contents/Developer`.
You can see it in the output of fastlane's environment:

```
Xcode Path | /Applications/Xcode.app/Contents/Developer/
```

You can also try setting the `DEVELOPER_DIR` to switch current Xcode.
For example, in shell session:

```bash
export DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer
```

or even in Ruby code before you call Xcode commands:

```ruby
ENV["DEVELOPER_DIR"] = "/Applications/Xcode.app/Contents/Developer"
```

**Comment by AndyIbanez** [comment]
@mgrebenets 

Thanks for your response.

Yes, the various equivalents do include the Xcode app with ".app" at the end and path to Developer. If you choose an invalid directory with `xcode-select`, the app complains that it is not a valid directory, so I know the variations I tried were correct.

But thanks for mentioning the last two options. It made me realize that I could have set something on my fastfile. And I in fact had this line:

`"xcode_select "/Applications/Xcode-beta.app"`

I have changed it now to `xcode_select "/Applications/Xcode.app"` and I will see if I can just delete the line.
